go/adbc/utils/utils.go

```diff
@@ -30,7 +30,15 @@ func RemoveSchemaMetadata(schema *arrow.Schema) *arrow.Schema {
 func removeFieldMetadata(field *arrow.Field) arrow.Field {
 	fieldType := field.Type
 
-	if nestedType, ok := field.Type.(arrow.NestedType); ok {
+	if ty, ok := field.Type.(*arrow.MapType); ok {
+		// XXX: this can't handle nonstandard field names or
```

Comment:
What was the actual issue that is being fixed?

Reply:
the original code thought a map directly had two children, instead we have to unpack it as it's actually a list[struct[

Reply:
Is there any particular reason to single out the map type like this instead of doing this in the switch case? The recursive call to `removeFieldMetadata` should handle calling removeFieldMetadata on the child of the list[struct]. We just need to modify the switch case appropriately to create the new map type without the metadata. Personally I think that would be cleaner than this entire separate if case.

Thoughts?

Reply:
You're not allowed to build a map type from the child field. So we have to special case it here.

Reply:
Updated to make it clearer what the issue is.